### PR TITLE
Fix CI worker tests for gated datasets

### LINKED ISSUE #3025: CI worker tests are broken for gated datasets: ConnectionError
CI worker tests are broken for gated datasets: https://github.com/huggingface/dataset-viewer/actions/runs/10319430999/job/28567795990
```
FAILED tests/job_runners/config/test_split_names.py::test_compute_split_names_from_streaming_response[gated-False-SplitNamesFromStreamingError-DatasetNotFoundError] - AssertionError: assert 'ConnectionError' == 'DatasetNotFoundError'
  
  - DatasetNotFoundError
  + ConnectionError
FAILED tests/job_runners/dataset/test_config_names.py::test_compute_splits_response[gated-False-ConfigNamesError-DatasetNotFoundError] - AssertionError: assert 'ConnectionError' == 'DatasetNotFoundError'
  
  - DatasetNotFoundError
  + ConnectionError
```

### PR DESCRIPTION
**Pull request by albertvillanova**
Fix CI worker tests for gated datasets.

Fix #3025.

### CODE REVIEW

File: services/worker/tests/job_runners/config/test_split_names.py [hunk 1]
```diff
@@ -173,7 +173,7 @@ def test_doesnotexist(
         # should we really test the following cases?
         # The assumption is that the dataset exists and is accessible with the token
         ("does_not_exist", False, "SplitNamesFromStreamingError", "DatasetNotFoundError"),
-        ("gated", False, "SplitNamesFromStreamingError", "DatasetNotFoundError"),
+        ("gated", False, "SplitNamesFromStreamingError", "ConnectionError"),  # See: huggingface/huggingface_hub#2457
```
Comment (severo) [inline services/worker/tests/job_runners/config/test_split_names.py:176]: Coool. Maybe

```suggestion
        ("gated", False, "SplitNamesFromStreamingError", "ConnectionError"),  # See: huggingface/datasets#7109
```

File: services/worker/tests/job_runners/dataset/test_config_names.py [hunk 2]
```diff
@@ -99,7 +99,7 @@ def test_compute_too_many_configs(
         # should we really test the following cases?
         # The assumption is that the dataset exists and is accessible with the token
         ("does_not_exist", False, "ConfigNamesError", "DatasetNotFoundError"),
-        ("gated", False, "ConfigNamesError", "DatasetNotFoundError"),
+        ("gated", False, "ConfigNamesError", "ConnectionError"),  # See: huggingface/huggingface_hub#2457
```
Comment (severo) [inline services/worker/tests/job_runners/dataset/test_config_names.py:102]:
```suggestion
        ("gated", False, "ConfigNamesError", "ConnectionError"),  # See: huggingface/datasets#7109
```

**Review by severo — APPROVED** [review]

File: services/worker/tests/job_runners/config/test_split_names.py (hunk 1 above)
Comment (albertvillanova) [reply services/worker/tests/job_runners/config/test_split_names.py]: Done.

File: services/worker/tests/job_runners/dataset/test_config_names.py (hunk 2 above)
Comment (albertvillanova) [reply services/worker/tests/job_runners/dataset/test_config_names.py]: Done.

**Pull request merged**